File: melosys-eessi-app/src/main/java/no/nav/melosys/eessi/controller/dto/Arbeidssted.java
package no.nav.melosys.eessi.controller.dto;

import lombok.Data;
import no.nav.melosys.eessi.service.sed.helpers.LandkodeMapper;

@Data
public class Arbeidssted {

    private String navn;
    private Adresse adresse;
    private boolean fysisk;
    private String hjemmebase;

    public static Arbeidssted av(no.nav.melosys.eessi.models.sed.nav.Arbeidssted arbeidsstedFraRina) {
        Arbeidssted arbeidssted = new Arbeidssted();

        arbeidssted.navn = arbeidsstedFraRina.getNavn();
        arbeidssted.adresse = Adresse.av(arbeidsstedFraRina.getAdresse());
        arbeidssted.fysisk = mapFysisk(arbeidsstedFraRina.getErikkefastadresse());
        arbeidssted.hjemmebase = LandkodeMapper.mapTilNavLandkode(arbeidsstedFraRina.getHjemmebase());

        return arbeidssted;
    }

    public static boolean mapFysisk(String erIkkefastadresse) {
        return "nei".equalsIgnoreCase(erIkkefastadresse);
    }
}

File: melosys-eessi-app/src/main/java/no/nav/melosys/eessi/controller/dto/Arbeidssted.kt
package no.nav.melosys.eessi.controller.dto

import no.nav.melosys.eessi.models.sed.nav.Arbeidssted as RinaArbeidssted
import no.nav.melosys.eessi.service.sed.helpers.LandkodeMapper

data class Arbeidssted(
    var navn: String? = null,
    var adresse: Adresse, // kaster NullPointerException i Java kode om null
    var fysisk: Boolean = false,
    var hjemmebase: String? = null
) {
    companion object {
        fun av(arbeidsstedFraRina: RinaArbeidssted): Arbeidssted = Arbeidssted(
            navn = arbeidsstedFraRina.navn,
            adresse = Adresse.av(arbeidsstedFraRina.adresse),
            fysisk = mapFysisk(arbeidsstedFraRina.erikkefastadresse),
            hjemmebase = LandkodeMapper.mapTilNavLandkode(arbeidsstedFraRina.hjemmebase)
        )

        private fun mapFysisk(erIkkefastadresse: String?): Boolean =
            erIkkefastadresse.equals("nei", ignoreCase = true)
    }
}
